Update dd-trace-go version reqs (#33128)
DataDog/documentation: content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/go.md

## Compatibility requirements

> The Go Tracer follows the [Official Go release policy](https://go.dev/doc/devel/release#policy) so only the latest two Go versions are supported. It also requires the Datadog Agent `>= 5.21.1`. For a full list of Datadog's Go version and framework support (including legacy and maintenance versions), see the [Compatibility Requirements][1] page.

Reviewer: ```suggestion
The Go Tracer follows the [Official Go release policy](https://go.dev/doc/devel/release#policy) and supports only the two most recent Go versions. It also requires the Datadog Agent `>= 5.21.1`. For a full list of Datadog's Go version and framework support (including legacy and maintenance versions), see the [Compatibility Requirements][1] page.
```
